iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -24,6 +24,62 @@
 namespace mlir {
 namespace iree_compiler {
 
+void ConvertToDynamicSharedMemory(ModuleOp moduleOp) {
+  // Collect all the adressOfOps to static shared memory globals.
+  SmallVector<LLVM::AddressOfOp> addressOfOps;
+  moduleOp.walk([&](LLVM::AddressOfOp addressOfOp) {
+    if (!addressOfOp.getGlobal().getConstant())
```

Comment:
I think we should check that the address space is shared memory.

Reply:
So have a check for addres_space == 3?

Reply:
Yes